packages/playwright/src/worker/workerMain.ts

```diff
@@ -75,25 +75,16 @@ export class WorkerMain extends ProcessRunner {
 
     process.on('unhandledRejection', reason => this.unhandledError(reason));
     process.on('uncaughtException', error => this.unhandledError(error));
-    process.stdout.write = (chunk: string | Buffer) => {
-      const outPayload: TestOutputPayload = {
-        ...chunkToParams(chunk)
-      };
-      this.dispatchEvent('stdOut', outPayload);
-      this._currentTest?._tracing.appendStdioToTrace('stdout', chunk);
-      return true;
-    };
+  }
```

Comment:
I think it would be acceptable to intercept the write and send output of this process to tracing, while feeding the reporters off the worker process output as you did. It'll fix one half of the issue.